_posts/2019-03-18-rules-jvm-external-maven.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: posts
+title: "A repository rule for calculating transitive Maven dependencies"
+authors:
+  - jin
+  - dkelmer
+---
+
+We’re delighted to introduce the initial release of
+[`rules_jvm_external`](https://github.com/bazelbuild/rules_jvm_external), a
+repository rule to resolve and fetch artifacts transitively in Maven
+repositories, like Maven Central and Google Maven. This project is officially
+maintained by the Bazel team.
+
+## Background and motivation
+
+Maven integration in Bazel is a large space with many existing robust solutions
+from community members:
+
+* [https://github.com/johnynek/bazel-deps](https://github.com/johnynek/bazel-deps)
+    * Tool to generate Bazel dependencies transitively for Maven artifacts, with
+      Scala and Kotlin support
+* [https://github.com/square/bazel\_maven\_repository](https://github.com/square/bazel_maven_repository)
+    * Repository rule to create an idiomatic Bazel representation of a Maven
+      repository using a pinned list of artifacts, with support for Kotlin and
+      Android dependencies
+* [https://github.com/menny/mabel](https://github.com/menny/mabel)
+    * Tool to generate a Maven dependency graph as a lock file, with support for
+      Kotlin and Android dependencies
+
+Each project approaches the problem slightly differently. Check out those
+projects as well to see their full feature set and determine which solution
+works best for you. It is also possible to wire dependencies up manually using
+rules like `java_import_external` and `aar_import_external`.
```

Comment:
Do we want to keep and maintain 4 related projects?
Is rules_jvm_external more important than the others (to deserve a blog post)?

Reply:
Which 4? 

gmaven_rules is replaced by rules_jvm_external. java_import_external and aar_import_external are manual repository rules for users to wire up Maven artifacts themselves.

rules_jvm_external is *the* officially supported Maven resolver tool for Bazel, going forward.

Reply:
`java_import_external` and friends will also be merged into `rules_jvm_external`, but those will need to go through an incompatible change process.

Reply:
Right, first of all we actually do not maintain other projects; 

Secondly, `rules_jvm_external` is a _potential_ replacement to them all but we are not deprecating or removing the other ones - I really like that this blog post starts with the homage to the all other projects that were developed by our community.

Reply:
Ah... I completely misunderstood @laurentlb's initial comment. I was thinking of gmaven_rules, rules_jvm_external, java_import_external and aar_import_external.